app/components/open_project/common/check_all_component.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+# -- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+# ++
+#
+module OpenProject
+  module Common
+    class CheckAllComponent < ApplicationComponent
+      include Primer::AttributesHelper
+
+      attr_reader :checkable_id
+
+      CHECKABLE_CONTROLLER_SELECTOR = "[data-controller~='checkable']"
+
+      renders_one :check_all, ->(text: I18n.t(:button_check_all), **system_arguments) {
+        action = use_outlet? ? "check-all#checkAll:stop" : "checkable#checkAll:stop"
+        controls = checkable_id if use_outlet?
+
+        system_arguments[:data] = merge_data(
+          system_arguments, {
+            data: { action: }
+          }
+        )
+        system_arguments[:aria] = merge_aria(
```

Comment:
aria-controls might be empty or invalid here. When use_outlet? is false, controls is nil, aria-controls="".
Empty ARIA attributes are invalid.

It can be:

```
if use_outlet?
  system_arguments[:aria] = merge_aria(system_arguments, { aria: { controls: checkable_id } })
end
```